ts/screens/onboarding/TosScreen.tsx

```diff
@@ -84,4 +111,18 @@ class TosScreen extends React.PureComponent<Props, State> {
   };
 }
 
-export default connect()(TosScreen);
+// If the user had accepted a previous version of ToS
+// it is displayed a related alert
+function mapStateToProps(state: GlobalState) {
+  const potProfile = profileSelector(state);
+  return {
+    hasPreviousAcceptedToS:
+      pot.isSome(potProfile) &&
+      potProfile.value.has_profile &&
+      (("accepted_tos_version" in potProfile.value &&
+        potProfile.value.accepted_tos_version !== tosVersion) ||
+        !("accepted_tos_version" in potProfile.value))
```

Comment:
So if the user has no "accepted_tos_version" property in the profile you are returning hasPreviousAcceptedToS = true
I don't think it is appropriate.

Reply:
Yes, but only if the user has an existing profile: this is the case the user started using the app before the accepted_tos_version property was introduced on backend profile. So, he previously accepted ToS even if the attribute is not stored on his profile.
Do you mind a more appropriate name for this variable?
cc @fpersico

Reply:
what about hasUserAlreadyAcceptedToS ?